benchmark/Main.hs

```diff
@@ -1,47 +1,86 @@
 
+{-# INLINE _64KB #-}
+_64KB :: Int
+_64KB = 64 * 1024
+
 --------------------------------------------------------------------------------
 -- Corpora helpers
 --------------------------------------------------------------------------------
 
 {-# INLINE large_bible_txt #-}
-large_bible_txt :: String -> String
-large_bible_txt base = base ++ "large/bible.txt"
+large_bible_txt :: String
+large_bible_txt = "large/bible.txt"
 
 {-# INLINE large_world192_txt #-}
-large_world192_txt :: String -> String
-large_world192_txt base = base ++ "large/world192.txt"
+large_world192_txt :: String
+large_world192_txt = "large/world192.txt"
 
 {-# INLINE cantrbry_alice29_txt #-}
-cantrbry_alice29_txt :: String -> String
-cantrbry_alice29_txt base = base ++ "cantrbry/alice29.txt"
+cantrbry_alice29_txt :: String
+cantrbry_alice29_txt = "cantrbry/alice29.txt"
+
+--------------------------------------------------------------------------------
+-- Bootstrapping
+--------------------------------------------------------------------------------
 
-{-# INLINE cantrbry_kennedy_xls #-}
-cantrbry_kennedy_xls :: String -> String
-cantrbry_kennedy_xls base = base ++ "cantrbry/kennedy.xls"
+normalizedName :: String -> String
+normalizedName x = x ++ ".normalized"
+
+bigCompressedName :: String -> String
+bigCompressedName x = x ++ ".normalized.compressed.big"
+
+smallCompressedName :: String -> String
+smallCompressedName x = x ++ ".normalized.compressed.small"
+
+bootstrap :: String -> IO ()
+bootstrap file = do
+    base <- getCurrentDirectory
+    let fp = base ++ "/corpora/" ++ file
+        normalizedFp = normalizedName fp
+        compressedFpBig = bigCompressedName fp
+        compressedFpSmall = smallCompressedName fp
+    fileExists <- doesFileExist normalizedFp
+    unless fileExists $ do
+        size <- withFile normalizedFp ReadMode hFileSize
+        let additionalLoops = _64KB `div` (fromIntegral :: Integer -> Int) size
+            fileStream =
+                Stream.unfold File.readChunksWithBufferOf (64 * 1024, fp)
+                    & pack64KB
+            -- 160 * 64 * 1024 = 10MB
+            combinedStream =
+                Stream.take 160
+                    $ foldr (Stream.append) fileStream
+                    $ replicate additionalLoops fileStream
+        combinedStream & File.fromChunks normalizedFp
+        combinedStream & LZ4.compress 65537 & File.fromChunks compressedFpBig
+        combinedStream & LZ4.compress 1 & File.fromChunks compressedFpSmall
 
-{-# INLINE artificl_random_txt #-}
-artificl_random_txt :: String -> String
-artificl_random_txt base = base ++ "artificl/random.txt"
+    where
 
-{-# INLINE artificl_aaa_txt #-}
-artificl_aaa_txt :: String -> String
-artificl_aaa_txt base = base ++ "artificl/aaa.txt"
+    pack64KB
+        :: Stream.SerialT IO (Array.Array Word8)
+        -> Stream.SerialT IO (Array.Array Word8)
+    pack64KB =
+        StreamD.fromStreamD . Array.packArraysChunksOf _64KB . StreamD.toStreamD
```

Comment:
`packArraysChunksOf` does not guarantee exact size. You should flatten the stream and use `arraysOf`.